src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>

```

Comment:
Also this can be done in different classes with different means to get a similar result. I don't have a problem refactoring this into something else.
It works with `CompileToMethod()`'s restrictions and doesn't look ugly as sin, so I left it like this.